### DISCUSSION TITLE
What chances are given to combine two-factor-authentication with EMQX?

### DISCUSSION THREAD

**Discussion by neilyoung** [emqx/emqx#12967] [General]
... as subject says

**Comment by thalesmg** [comment]
Hi.

Do you mean 2FA for dashboard access?

**Comment by neilyoung** [comment]
Yes, that would be perfect to have

**Comment by thalesmg** [comment]
We already support SAML SSO: https://docs.emqx.com/en/enterprise/latest/dashboard/sso-saml.html#configure-saml-based-sso

2FA may then be implemented by the SSO provider.

**Reply by neilyoung** [reply]
Thanks, requires enterprise edition, I guess..

**Reply by zmstone** [reply]
yes, this feature is for Enterprise edition

**Reply by kevdogg** [reply]
Run the dashboard through a reverse proxy that forwards to authelia of your 2FA method of choice.
